worker/container_sweeper_test.go

```diff
@@ -0,0 +1,153 @@
+package worker_test
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+	"time"
+
+
+	"github.com/concourse/concourse/worker/workerfakes"
+
+	"code.cloudfoundry.org/lager/lagertest"
+	"github.com/concourse/concourse/atc/worker/gclient"
+	"github.com/concourse/concourse/worker"
+	"github.com/onsi/gomega/ghttp"
+
+	"github.com/onsi/gomega/gbytes"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Container Sweeper", func() {
+	const (
+	 sweepInterval              = 50 * time.Millisecond
+	 maxInFlight                =  uint16(1)
+	 gardenClientTimeoutRequest = 5 * time.Millisecond
+	)
+
+	var (
+		garden *ghttp.Server
+
+		testLogger = lagertest.NewTestLogger("container-sweeper")
+
+		fakeTSAClient workerfakes.FakeTSAClient
+		gardenClient  gclient.Client
+
+		sweeper *worker.ContainerSweeper
+
+		osSignal  chan os.Signal
+		exited    chan struct{}
+
+	)
+
+	BeforeEach(func() {
+		garden = ghttp.NewServer()
+
+		osSignal = make(chan os.Signal)
+		exited = make(chan struct{})
+
+		gardenAddr := fmt.Sprintf("http://%s", garden.Addr())
+		gardenClient = gclient.BasicGardenClientWithRequestTimeout(testLogger, gardenClientTimeoutRequest, gardenAddr)
+
+		fakeTSAClient = workerfakes.FakeTSAClient{}
+		fakeTSAClient.ReportContainersReturns(nil)
+
+		sweeper = worker.NewContainerSweeper(testLogger, sweepInterval, &fakeTSAClient, gardenClient, maxInFlight)
+
+	})
+
+	JustBeforeEach(func() {
+		go func() {
+			_ = sweeper.Run(osSignal, make(chan struct{}))
+			close(exited)
+		}()
+	})
+
+	AfterEach(func() {
+		close(osSignal)
+		<-exited
+		garden.Close()
+	})
+
+	Context("when garden doesn't respond on DELETE", func() {
+		var (
+			gardenContext context.Context
+			gardenCancel  context.CancelFunc
+		)
+
+		BeforeEach(func() {
+			gardenContext, gardenCancel = context.WithCancel(context.Background())
+			garden.AppendHandlers(
+				ghttp.CombineHandlers(
+					ghttp.VerifyRequest("GET", "/containers"),
+					ghttp.RespondWithJSONEncoded(200, []map[string]string{}),
+				),
+				ghttp.CombineHandlers(
```

Comment:
nit: I think we could separate the way that handlers are set up here - if the only thing that this `Context()` changes is the `DELETE` handlers, I think it'd be good to have just those handlers set up at this level. Wdyt?

E.g.:

```
Describe("containersweeper", func () {

  BeforeEach(func () {
      // setup of handlers that are needed by everyone
  }) 

  Context("when DELETEs never return", func () {

        BeforeEach(func () {
             // configure the DELETE handlers.
        })

   })
})
``` 

wdyt?

Reply:
+1

Reply:
As the ghttp server takes a ordered queue of handlers, it would be awkward to split this.